kernel/packages/shared/index.ts

```diff
@@ -9,29 +9,33 @@ import { loadingStarted, notStarted, MOBILE_NOT_SUPPORTED } from './loading/type
 import { buildStore } from './store/store'
 import { initializeUrlPositionObserver } from './world/positionThings'
 import { StoreContainer } from './store/rootTypes'
-import { login } from './session/actions'
+import { initSession, login } from './session/actions'
+import { ENABLE_WEB3, PREVIEW } from '../config'
+import { ProviderType } from './ethereum/ProviderType'
 
 declare const globalThis: StoreContainer
 
 export function initShared() {
+  if (globalThis.globalStore) {
```

Comment:
does this mean that now initShared can be called multiple times?

Reply:
No, it is called only one time, the difference is that in react app we need to have the redux store before `initShared` is called then here we check it.